src/Instrumentation/Doctrine/src/DoctrineInstrumentation.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Doctrine;
+
+use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanBuilderInterface;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+
+use function OpenTelemetry\Instrumentation\hook;
+
+use OpenTelemetry\SemConv\TraceAttributes;
+use Throwable;
+
+class DoctrineInstrumentation
```

Comment:
You should call this DBAL instrumentation. Possibly whole package as well. Have you thought about Doctrine ORM? Is that going to be a separate package, or part of this package? If the former, this package should be renamed. If the latter, code should be structured in a way to be ready for future extension to cover doctrine ORM. I'm pretty sure it will be important because object hydration is often the bottleneck.

Reply:
Well I think it's fine to have this umbrella-instrumentation. I'd be happy to add some more ORM instrumentation later, but I think it makes sense to stick with this naming as ORM depends on DBAL

Reply:
Ok. Just a note though: doctrine/orm depends on doctrine/dbal, but not doctrine/mongodb-odm

Reply:
> You should call this DBAL instrumentation. Possibly whole package as well. Have you thought about Doctrine ORM? Is that going to be a separate package, or part of this package? If the former, this package should be renamed. If the latter, code should be structured in a way to be ready for future extension to cover doctrine ORM. I'm pretty sure it will be important because object hydration is often the bottleneck.

Right. I didn't think at the possibility to monitor Doctrine ORM Hydration process., only the database invocations.




> Well I think it's fine to have this umbrella-instrumentation. I'd be happy to add some more ORM instrumentation later, but I think it makes sense to stick with this naming as ORM depends on DBAL

I really like that idea. I will maintain the current nomenclature